addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4238,7 +4238,13 @@ var SnippetsMenu = Widget.extend({
      * @param {function} ev.data.exec
      */
     _onSnippetEditionRequest: function (ev) {
-        this._execWithLoadingEffect(ev.data.exec, true);
+        if (ev.data.optionsLoader) {
+            this._execWithLoadingEffect(() => {
+                this._execWithLoadingEffect(ev.data.exec, true);
+            }, false);
+        } else {
+            this._execWithLoadingEffect(ev.data.exec, true);
+        }
```

Comment:
In your commit message you say "when a change is made in one of the colorpickers of the
"Options tab". These aren't actually all the colorpickers in the "Options tab", they're only 3 colorpickers in the "Colors" section (those with gradient available). If I change the color of the "Input Fields" or "Status Colors", I don't see the loader. Why do we need it for the “Colors” section and not elsewhere, even for option like "Page Layout" for example? And if we want to go further, why wouldn't we also need a loader for any options in the "Blocks" tab? I don't understand why we need it more for the part where you added it compared to other places.

Reply:
Adapted the commit message to mention "the colorpickers that give access to gradients" instead.
Those are the ones that need this feature, because the update of the SCSS drives the update of the gradient editor, so if you can continue to use it without waiting for the update, it messes up the requested changes.
If you want to see the kind of mess: remove that commit, then start adding gradient splits and changing colors - you will see the content of the editor being updated to older versions of the gradient while you already were much further.
Other options do not have an editor panel that remains opened during the update - you always end up with a single plain value. It might get reset, but it is less messy.